test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/EdmDateAndTimeOfDayTests/Tests/EdmDateAndTimeOfDayTests.cs

```diff
@@ -0,0 +1,501 @@
+            var responseMessage = await requestMessage.GetResponseAsync();
+
+            Assert.Equal(200, responseMessage.StatusCode);
+
+            using var messageReader = new ODataMessageReader(responseMessage, readerSettings, _model);
+            var date = (await messageReader.ReadPropertyAsync()).Value;
+            Assert.Equal(new Date(2014, 8, 31), date);
+        }
+
+        [Fact]
+        public async Task InvokingAFunctionWithDateParameter_WorksCorrectly()
+        {
+            ODataMessageReaderSettings readerSettings = new() { BaseUri = _baseUri };
+
+            var requestUrl = new Uri(_baseUri.AbsoluteUri + "Orders(7)/Default.CheckShipDate(date = 2014-08-31)", UriKind.Absolute);
+
+            var requestMessage = new TestCommon.Common.HttpClientRequestMessage(requestUrl, Client);
+            requestMessage.SetHeader("Accept", "*/*");
+            var responseMessage = await requestMessage.GetResponseAsync();
+
+            Assert.Equal(200, responseMessage.StatusCode);
+
+            using (var messageReader = new ODataMessageReader(responseMessage, readerSettings, _model))
+            {
+                var True = (await messageReader.ReadPropertyAsync()).Value;
+                Assert.Equal(true, True);
+            }
+        }
+
+        [Fact]
+        public async Task InvokingAFunctionWithTimeReturnType_CorrectlyWorks()
+        {
+            ODataMessageReaderSettings readerSettings = new() { BaseUri = _baseUri };
+
+            var requestUrl = new Uri(_baseUri.AbsoluteUri + "Orders(7)/Default.GetShipTime", UriKind.Absolute);
+
+            var requestMessage = new TestCommon.Common.HttpClientRequestMessage(requestUrl, Client);
+            requestMessage.SetHeader("Accept", "*/*");
+            var responseMessage = await requestMessage.GetResponseAsync();
+
+            Assert.Equal(200, responseMessage.StatusCode);
+
+            using var messageReader = new ODataMessageReader(responseMessage, readerSettings, _model);
+            var time = (await messageReader.ReadPropertyAsync()).Value;
+
+            Assert.Equal(new TimeOfDay(12, 40, 5, 50), time);
+        }
+
+        [Fact]
+        public async Task InvokingAFunctionWithATimeParameter_WorksCorrectly()
+        {
+            ODataMessageReaderSettings readerSettings = new() { BaseUri = _baseUri };
+
+            var requestUrl = new Uri(_baseUri.AbsoluteUri + "Orders(7)/Default.CheckShipTime(time = 12:40:5.5)", UriKind.Absolute);
+
+            var requestMessage = new TestCommon.Common.HttpClientRequestMessage(requestUrl, Client);
+            requestMessage.SetHeader("Accept", "*/*");
+            var responseMessage = await requestMessage.GetResponseAsync();
+
+            Assert.Equal(200, responseMessage.StatusCode);
+
+            using var messageReader = new ODataMessageReader(responseMessage, readerSettings, _model);
+            var True = (await messageReader.ReadPropertyAsync()).Value;

```

Comment:
```suggestion
            var result = (await messageReader.ReadPropertyAsync()).Value;
```
`result` should be better than `True` - local variable in uppercase